src/core/dep-graph.ts

```diff
@@ -325,7 +325,7 @@ class DepGraphImpl implements types.DepGraphInternal {
       .map((parentId) => {
         const pathsFromParent = this.pathsFromNodeToRoot(
           parentId,
-          new Set(seenOnPath).add(nodeId),
+          new Set(seenOnPath).add(nodeId), // Max size is O(log(n)) [tree height] where n is the amount of nodes in the tree
```

Comment:
maybe "tree depth" is a more common term?

Reply:
I guess it depends on the school :) 
will update it